.buildkite/commands/install-node-dependencies.sh

```diff
@@ -0,0 +1,42 @@
+#!/bin/bash -eu
+
+PLATFORM=$(uname -s)
+ARCHITECTURE=$(uname -m)
+NODE_VERSION=$(node --version)
+PACKAGE_HASH=$(hash_file package-lock.json)
+GUTENBERG_PACKAGE_HASH=$(hash_file gutenberg/package-lock.json)
+JETPACK_PACKAGE_HASH=$(hash_file jetpack/pnpm-lock.yaml)
+CACHEKEY="$BUILDKITE_PIPELINE_SLUG-npm-$PLATFORM-$ARCHITECTURE-node$NODE_VERSION-$PACKAGE_HASH-$GUTENBERG_PACKAGE_HASH"
+PNPM_CACHEKEY="$BUILDKITE_PIPELINE_SLUG-pnpm-$PLATFORM-$ARCHITECTURE-node$NODE_VERSION-$JETPACK_PACKAGE_HASH"
+
+echo "--- :npm: Restore cache if present"
+restore_cache "$CACHEKEY"
+restore_cache "$PNPM_CACHEKEY"
+
+echo "--- :npm: Install Node dependencies"
+npm ci --unsafe-perm --prefer-offline --no-audit --no-progress "$@"
+
+echo "--- :npm: Save cache if necessary"
+# Notice that we don't cache the local node_modules.
+# Those get regenerated by npm ci as per Node reccomendations.
+# What we are caching is the root npm folder, which stores pacakge downloads so they are available if the package.json resolution demands them.
+save_cache "$HOME/.npm" "$CACHEKEY"
+
+# If we attempted to save the pnpm cache when npm run with '--prefix gutenber', the command might fail.
+# That's because the Jetpack submodule alone uses pnpm.
+# Therefore, if no pnpm cache was available at the start, running with '--prefix gutenberg' would not have generated one and 'save_cache' would fail to find it.
+if echo "$@" | grep -q -- "--prefix gutenber"; then
```

Comment:
I know this is still in draft but I was curious about this work 😄. Friendly note that there is a typo in this line, just in case you missed it and it might affect your testing.

Reply:
Thanks @geriux . I had missed it indeed!